### PR TITLE
Added rudimentary spellchecking with aspell

### PR DESCRIPTION
**Pull request by mdnahas**
You can now run "make spell" and you'll get a list of misspelled words.

The file tex/sorted_words_to_ignore.txt has a list of words that should not be reported as spelling errors.    Many are math words ("Abelian", "arity"), or Coq words ("bool") and some are LaTeX words ("belowskip").

THE LIST OF IGNORED WORDS NEEDS TO BE CHECKED BY SOMEONE.  I created it by generating all words disliked by aspell and then identifying the ones that I thought were mistakes.  I don't know all the words used internal to the document, so I only chose the most obvious misspelled words.  

Future: It might be worth splitting the files into separate Math, Coq, and LaTeX word lists.   We could also create an aspell dicationary file, but I thought that was too much work.   We could add a git hook that requires that any commit do not increase the list of misspelled words.

Currently, running "make spell" reports:   
[NOTE: Prenex is a valid math word, but I'm not sure it is necessary.  Adaptor is a word, but not the preferred spelling.  Learnt is a word, but archaic.]

Prenex
adaptor
archimedean
arithmetics
beign
cartesian
cauchy
chinese
coercions
datastructures
emacs
english
fniite
interveen
learnt
machanism
maching
materialise
matrixes
mentionned
nontriviality
ommit
orderedring
ordinaux
organisation
organised
organising
parametrize
premisse
prenex
realised
refering
resoning
swiss
tantamounts
toplevel

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -15,3 +15,6 @@ artwork:
 
 clean:
 	rm book.pdf ch*.html cover-front-web.png
+
+spell:
+	for f in ../tex/*.tex; do cat $$f | aspell -t list; done | sort | uniq | comm -2 -3 - ../tex/sorted_words_to_ignore.txt
```
Comment (anton-trunov) [inline docs/Makefile]: Please consider making `aspell` invocation locale-independent.
For example, on my machine `make spell` results in
```
Error: No word lists can be found for the language "ru_RU".
```

Setting the `--master` option to `en_US` fixes it for me:
```bash
aspell --master=en_US -t list
```

Comment (mdnahas) [inline docs/Makefile]: Done.  Pull request shows the new version, with 2 commits.

**Comment by amahboubi** [comment]
Thanks for this device @mdnahas. I am merging it and will discuss with the other authors the possible extensions you suggest.

**Comment by anton-trunov** [comment]
@mdnahas The git hook idea is really creative! Perhaps it'd be helpful to create a separate issue for it?

**Comment by amahboubi** [comment]
Thanks for the suggestion @anton-trunov: I have opened a separated issue. If anyone wishes to make further comments on this topic, please do so at #14.

**Pull request merged**